settings.gradle

```diff
@@ -1,48 +1,41 @@
-/*
- * This settings file was generated by the Gradle 'init' task.
- *
- * The settings file is used to specify which projects to include in your build.
- * In a single project build this file can be empty or even removed.
- *
- * Detailed information about configuring a multi-project build in Gradle can be found
- * in the user guide at https://docs.gradle.org/4.0/userguide/multi_project_builds.html
- */
-
 def includeSubprojectPerVersion(major,flavor,includeLegacy) {
-
+    
     new File("$rootProject.projectDir/2repository")
-
-            .listFiles({ File version -> version.name.startsWith(major)
-            } as FileFilter)
-
-            .collectMany ({ File versionflavor ->
-                versionflavor.listFiles({ File file -> file.name.contains(flavor)
-                } as FileFilter).toList()
-            })
-
-            .each { File repositoryFile ->
+    
+        .listFiles({ File version -> version.name.startsWith(major)
+		   } as FileFilter)
+    
+        .collectMany ({ File versionflavor -> 
+			versionflavor.listFiles({ File file -> file.name.contains(flavor)
+						} as FileFilter).toList()
+		      })
+    
+        .each { File repositoryFile ->
                 def version = repositoryFile.parentFile.name
                 def revision = repositoryFile.name
                 def projectSkeleton = "1skeleton:${version}"
                 def projectRepository = "2repository:${revision}"
                 def projectLegacy = "3legacy:${revision}"
-
-                if(flavor.equals("community")) {
-                    println "Including $projectSkeleton"
-                    include projectSkeleton
-                    project(":${projectSkeleton}").projectDir = file("1skeleton/${version}")
-                }
-                
-                println "Including $projectRepository"
-                include projectRepository
-                project(":${projectRepository}").projectDir = file("2repository/${version}/${revision}")
-
-                if (includeLegacy && new File("3legacy/${version}/${revision}").exists()) {
-                    println "Including $projectLegacy"
-                    include projectLegacy
-                    project(":${projectLegacy}").projectDir = file("3legacy/${version}/${revision}")
-                }
-            }
+		
+		if(new File("$repositoryFile/overload.gradle").exists()) {
```

Comment:
I do not understand why this check for `overload.gradle` is necessary.

Maybe because the changes that you need it for are in this PR, but can you explain?

Reply:
The explanation is in the ticket on Xenit's Jira: https://xenitsupport.jira.com/browse/DOCKER-372

```
Current way of deprecating a version is renaming the folder:

alfresco-search-services-community-1.3.1 => alfresco-search-services-com-rename-for-build-1.3.1

The build on jenkins fails because folder alfresco-search-services-community-1.3.1 will still exist (with an empty subfolder “build” inside)
```

Reply:
I hate it. :rage: 

But given that the other solution is to use `cleanWs()` to completely wipe out the Jenkins workspace for the project, I think this is better from a build-cache point of view.

Unless there is a third solution that we did not think about, I guess we have to go with this.

Can you please add a comment to explain *why* this check is there? It's not clear that this is there to catch directories that are left behind when a subproject is renamed.

Reply:
I amended the commit message.

Reply:
Please add a comment as well.

A whitespace change on this line, or a minor edit will make the commit message no longer discoverable with `git blame`.

It is very likely that the next person looking at it will be wondering why an `if` that always seems to be true is present, and will remove it, reintroducing this issue again.

Reply:
No problem, I added the comment. Communication is important.